src/main/java/org/jboss/pnc/common/http/PNCHttpClient.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2021 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.common.http;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import dev.failsafe.Failsafe;
+import dev.failsafe.RetryPolicy;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.jboss.pnc.api.dto.Request;
+
+import java.io.IOException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.function.Supplier;
+
+import static org.jboss.pnc.api.constants.HttpHeaders.AUTHORIZATION_STRING;
+
+/**
+ * Wrapper around {@link HttpClient} that can handle PNC {@link Request}s.
+ */
+@Slf4j
+public class PNCHttpClient {
+    private final HttpClient client;
+    private final ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
Don't we wanna make the `objectMapper` a little more robust?
IIRC, it's unable to serialize e.g. `LocalDateTime`, since the corresponding module needs to be registered when creating new instance of `ObjectMapper`